deepmd/dpmodel/model/linear_model.py

```diff
@@ -0,0 +1,234 @@
+        alpha: Optional[float] = 0.1,
+        mapping: Optional[np.ndarray] = None,
+    ) -> Dict[str, np.ndarray]:
+        """Return atomic prediction.
+
+        Note: currently only support the linear combination of a ZBL model and a DP model,
+        the weight is calculated based on this paper:
+        Appl. Phys. Lett. 114, 244101 (2019); https://doi.org/10.1063/1.5098061
+
+        Parameters
+        ----------
+        extended_coord
+            coodinates in extended region, (nframes, nall * 3)
+        extended_atype
+            atomic type in extended region, (nframes, nall)
+        nlist
+            neighbor list, (nframes, nloc, nsel).
+        mapping
+            mapps the extended indices to local indices
+        ra : float
+            inclusive lower boundary of the range in which the ZBL potential and the deep potential are interpolated.
+        rb : float
+            exclusive upper boundary of the range in which the ZBL potential and the deep potential are interpolated.
+        alpha : float
+            a tunable scale of the distances between atoms.
+
+        Returns
+        -------
+        result_dict
+            the result dict, defined by the fitting net output def.
+        """
+        nframes, nloc, nnei = nlist.shape
+        extended_coord = extended_coord.reshape(nframes, -1, 3)
+        nlists = build_multiple_neighbor_list(
+            extended_coord,
+            nlist,
+            [self.zbl_model.rcut, self.dp_model.get_rcut()],
+            [self.zbl_model.sel, sum(self.dp_model.get_sel())],
+        )
+        zbl_nlist = nlists[str(self.zbl_model.rcut) + "_" + str(self.zbl_model.sel)]
+        dp_nlist = nlists[
+            str(self.dp_model.get_rcut()) + "_" + str(sum(self.dp_model.get_sel()))
+        ]
+
+        zbl_nnei = zbl_nlist.shape[-1]
+        dp_nnei = dp_nlist.shape[-1]
+
+        # use the larger rr based on nlist
+        nlist_ = zbl_nlist if zbl_nnei >= dp_nnei else dp_nlist
+        masked_nlist = np.clip(nlist_, 0, None)
+        pairwise_rr = np.sqrt(
+            np.sum(
+                np.power(
+                    (
+                        np.expand_dims(extended_coord, 2)
+                        - np.expand_dims(extended_coord, 1)
+                    ),
+                    2,
+                ),
+                axis=-1,
+            )
+        )
+
+        rr = np.take_along_axis(pairwise_rr[:, :nloc, :], masked_nlist, 2)
+        # (nframes, nloc, 1)
+        self.zbl_weight = self._compute_weight(nlist_, rr, ra, rb, alpha)
+        # (nframes, nloc, 1)
+        dp_energy = self.dp_model.forward_atomic(
+            extended_coord, extended_atype, dp_nlist
+        )["energy"]
+        # (nframes, nloc, 1)
+        zbl_energy = self.zbl_model.forward_atomic(
+            extended_coord, extended_atype, zbl_nlist
+        )["energy"]
+
+        fit_ret = {
+            "energy": (self.zbl_weight * zbl_energy + (1 - self.zbl_weight) * dp_energy)
```

Comment:
We might need to add the bias term here. It seems on `tf` side, the bias is calculated from data statistics `atom_bias_ener` in `Fitting`. I don't see a similar implementation in `pt`, should I leave it as an input parameter.

Reply:
Yes the data stat interface has not been added to the pt fittings. waiting for the design of @iProzd